src/modules/TodoList/component.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+import { Todo } from '../Todo';
+
+export const Component = ({ todos, finalMessage }) => (
+  <div>
+    {
+      Array.isArray(todos) && todos.length
+      ? todos.map((todo, index) => (
+          <Todo 
+            key={todo.id}
+            text={todo.text}
+            onClickDelete={() => 
+              deleteTodo(index)
```

Comment:
Can you tell me how the delete todo functionality is working here?

Reply:
I'm very sorry but i missed to pass the deleteTodo function as props to children when I decided at last to split the main component into smaller ones in order to create specific components for specific tasks.
Thank you for noticed that because it allowed me to also fix the deleteTodo function itself because I forgot that the Array.splice function wasn't pure functional.